### DISCUSSION TITLE
properly Editing Main and Secondary palettes to show underwater palette

### DISCUSSION THREAD

**Discussion by Aikocho** [TrianglyRU/OrbinautFramework#39] [Q&A] [answered]
Basically put, how should you edit palettes to show up properly in the level? I been experimenting with the scripts for character and stage palettes, however, the water palette doesn't seem to appear right. I don't know if I missed something. What are all the things I need to edit specifically?

![Runner_3m5snWAZ4Y](https://github.com/user-attachments/assets/8fcc705c-c302-4684-bcd3-9856c39b3e3d)

**Comment by TrianglyRU** [comment]
Sorry for the late answer! Could you please attach your palette sheet here?

**Comment by Aikocho** [comment]
![1](https://github.com/user-attachments/assets/b20c828b-7736-4dfb-bc8c-6e1d116c62a5)
![2](https://github.com/user-attachments/assets/b5d2fb2b-3b70-43eb-b6ef-981392d00eca)

Yesir, these are my current palette sheets.

**Comment by TrianglyRU** [comment, marked as answer]
The arrangement looks fine to me. Ensure the colour values of the original colours in the sheet match with the graphics of the stage. Looks like practically all stage colours do not match. For example, those greens here do not. You may compare them with your own screenshot you've sent before.

![изображение](https://github.com/user-attachments/assets/c2921cb8-71bd-42ff-932c-9e2da342db70)

Also, since you're extending the palette sheet, you need to shift the indexes used for palette rotations, for both the player and the stage. But the basic palette swap should work nonetheless.

**Comment by Aikocho** [comment]
I'm actually embarrassed now. But I get it. I really could barely see the difference, but I can do better now. I thank you for the answer man.

**Reply by TrianglyRU** [reply]
Glad to help!
